lib/net/ssh/known_hosts.rb

```diff
@@ -127,6 +131,21 @@ def keys_for(host)
       keys
     end
 
+    # Indicates whether one of the entries matches an hostname that has been
+    # stored as a HMAC-SHA1 hash in the known hosts.
+    def known_host_hash?(hostlist, entries, scanner)
+      if hostlist.size == 1 && hostlist.first =~ /\A\|1(\|.+){2}\z/
```

Comment:
Because when the hostname is hashed, `hostlist = host.split(/,/) = ["|1|eKp+6E0rZ3lONgsIziurXEnaIik=|rcQB/rlJMUquUyFta64KugPjX4o="]`